packages/devtools/lib/src/memory/memory_data_view.dart

```diff
@@ -0,0 +1,158 @@
+        }
+      } else if (value is Sentinel) {
+        valueStr = value.valueAsString;
+      } else if (value is TypeArgumentsRef) {
+        valueStr = value.name;
+      } else {
+        valueStr = value.toString();
+      }
+
+      final CoreElement element = li(c: 'memory-instance-data-list-item')
+        ..add([
+          span(text: name),
+          span(text: ' $valueStr', c: 'subtle'),
+        ]);
+
+      StreamSubscription sub;
+
+      sub = element.element.onMouseOver.listen((e) {
+        // TODO(devoncarew): Call toString() only after a short dwell.
+        sub.cancel();
+        variableDescriber(field).then((String desc) {
+          element.tooltip = desc;
+        });
+      });
+
+      return element;
+    });
+  }
+
+  MemoryController _memoryController;
+  SelectableTree<BoundField> _items;
+
+  List<BoundField> get items => _items.items;
+
+  @override
+  CoreElement get element => _items;
+
+  void showFields(List<BoundField> fields) {
+    // AsyncCausal frames don't have local vars.
+    _items.setItems(fields);
+  }
+
+  void clearFields() {
+    _items.setItems(<BoundField>[]);
+  }
+}
+
+class MemoryDataChildProvider extends ChildProvider<BoundField> {
+  MemoryDataChildProvider(this._memoryController);
+
+  final MemoryController _memoryController;
+
+  @override
+  bool hasChildren(BoundField item) =>
+      item.value is InstanceRef && item.value.valueAsString == null;
+
+  @override
+  Future<List<BoundField>> getChildren(BoundField item) async {
+    final BoundField field = item;
+
+    if (field.value != null && field.value is InstanceRef) {
+      switch (field.value.kind) {
+        case InstanceKind.kPlainInstance:
+          final Instance instance =
+              await _memoryController.getObject(field.value.id);
+          return instance.fields;
+          break;
+        case InstanceKind.kList:
+          final Instance instance =
+              await _memoryController.getObject(field.value.id);
+          final List<BoundField> result = [];
+
+          int index = 0;
+          for (dynamic value in instance.elements) {
+            result.add(new BoundField()
+              ..decl = (FieldRef()..name = '[$index]')
+              ..value = value);
+            index++;
+          }
+          return result;
+          break;
+        case InstanceKind.kMap:
+          final Instance instance =
+          await _memoryController.getObject(field.value.id);
+
+          final List<BoundField> result = [];
+          for (dynamic value in instance.associations) {
+            result.add(new BoundField()
+              // TODO(terry): Need to handle nested objects for keys/values.
+              ..decl = (FieldRef()..name = '[${value.key.valueAsString}]')
+              ..value = value.value.valueAsString);
+          }
+          return result;
+          break;
+        case InstanceKind.kStackTrace:
+          print('TODO(terry): Handle StackTrace type.');
```

Comment:
instead of a print, can you add a proper todo and display a placeholder value? Maybe just the toString?

Reply:
Yes